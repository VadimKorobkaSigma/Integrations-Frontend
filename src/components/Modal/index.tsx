import * as React from 'react';
import cn from 'classnames';

import styles from './style.module.scss';
import Button from '@components/Button';

interface Props {
    isOpen: boolean;
    header?: string | React.ReactElement;
    footer?: string | React.ReactElement;
    onClose: VoidFunction;
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    children: React.ReactElement<any, any> | React.ReactNodeArray;
}

const Modal: React.FC<Props> = ({ children, isOpen, onClose, header, footer }) => {
    return (
        <div className={cn(styles.modal, isOpen && styles.open)}>
            <div className={styles.backdrop} onClick={onClose} />
            <div className={styles.container}>
                <header className={styles.header}>
                    <h1>{header}</h1>
                </header>
                {children}
                {footer && (
                    <footer className={styles.footer}>
                        <Button variant="link" onClick={onClose}>
                            Cancel
                        </Button>
                        {footer}
                    </footer>
                )}
            </div>
        </div>
    );
};

export default React.memo(Modal);
