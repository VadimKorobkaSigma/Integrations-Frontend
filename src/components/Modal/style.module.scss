.modal {
  position: absolute;
  z-index: 1;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  pointer-events: none;

  .backdrop,
  .container {
    transition: opacity 0.6s;
    opacity: 0;
  }

  &.open {
    pointer-events: initial;

    .backdrop,
    .container {
      opacity: 1;
    }
  }

  .backdrop {
    background-color: #00000060;
    position: absolute;

    width: 100%;
    height: 100%;
  }

  .container {
    position: absolute;
    top: 10%;
    bottom: 10%;
    left: 25%;
    right: 25%;
    color: $text-black;
    background-color: $white;
    box-shadow: #00000012 0px 4px 10px;
    padding: 24px 32px;
    border-radius: 4px;
    display: grid;
    grid-template-rows: fit-content(0) 1fr fit-content(0);
    border-top: 4px solid #2991f3;
  }
}

.header {
  h1 {
    text-align: center;
    color: $grey-dark;
  }
}

.header {
  text-align: center;
  color: $grey-dark;
}

.footer {
  display: flex;
  justify-content: flex-end;

  & > *:not(:last-child) {
    margin-right: 1rem;
  }
}
