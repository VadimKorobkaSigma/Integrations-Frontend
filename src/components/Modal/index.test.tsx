import * as React from 'react';
import Modal from './index';
import { shallow } from 'enzyme';

test('Modal Component', () => {
    const wrapper = shallow(
        <Modal
            isOpen={true}
            onClose={() => {
                wrapper.setProps({ isOpen: false });
            }}
            footer={<footer>Footer</footer>}
        >
            <p>1</p>
        </Modal>,
    );
    expect(wrapper.html()).toMatchSnapshot();
});
