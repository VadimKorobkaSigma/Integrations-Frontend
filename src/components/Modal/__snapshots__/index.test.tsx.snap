// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component 1`] = `"<div class=\\"\\"><div></div><div><header><h1></h1></header><p>1</p><footer><button variant=\\"link\\" class=\\"\\">Cancel</button><footer>Footer</footer></footer></div></div>"`;
